src/App/Fig.purs

```diff
@@ -46,189 +45,280 @@ str =
 
-      inertBwd = vertices g0 \\ (vertices $ snd (gcBwd (topOf outα))) :: Set DVertex
-      inertFwd = vertices $ snd $ (graphgc.fwd <<< focusFwd) γ0
+      vf :: Val (SelState 𝔹) -> Env (SelState 𝔹) × GraphImpl
+      vf = lift inert'.γ gcBwd
 
-      γInert = selState <$> select𝔹s γα inertBwd :: Env (𝔹 -> 𝔹 -> SelStates 𝔹)
-      vInert = selState <$> select𝔹s outα inertFwd :: Val (𝔹 -> 𝔹 -> SelStates 𝔹)
+      γf :: Env (SelState 𝔹) -> Val (SelState 𝔹) × GraphImpl
+      γf = lift inert'.v gcFwd
 
-      vf = lift γInert gcBwd
-      γf = lift vInert gcFwd
+      ι_fwd :: Set Vertex -> Val (SelState 𝔹)
+      ι_fwd ια = inert'.v <*> (ι_fwd' ια)
 
-      linkedInputs :: Env (SelStates 𝔹) -> Env (SelStates 𝔹) × Val (SelStates 𝔹) × Selection GraphImpl × Set DVertex
-      linkedInputs = γf >>> \(v × g × g') -> (fst $ vf v) × v × { persistent: g, transient: g' } × inertFwd
+      ι_bwd :: Set Vertex -> Env (SelState 𝔹)
+      ι_bwd ια = inert'.γ <*> (ι_bwd' ια)
 
-      linkedOutputs :: Val (SelStates 𝔹) -> Val (SelStates 𝔹) × Env (SelStates 𝔹) × Selection GraphImpl × Set DVertex
-      linkedOutputs = vf >>> \(γ × g × g') -> (fst $ γf γ) × γ × { persistent: g, transient: g' } × inertBwd
+      linkedInputs :: SelectionType -> Env (SelStates 𝔹) -> Env (SelState 𝔹) × Val (SelState 𝔹) × Set DVertex
+      linkedInputs selType γ =
+         let v × g = γf (γ <#> getSel selType) in fst (vf v) × v × (vertices g)
+
+      linkedOutputs :: SelectionType -> Val (SelStates 𝔹) -> Env (SelState 𝔹) × Val (SelState 𝔹) × Set DVertex
+      linkedOutputs selType v =
+         let γ × g = vf (v <#> getSel selType) in γ × fst (γf γ) × (vertices g)
+
+      linkIntermediates :: Env (SelStates 𝔹) -> Env (SelState 𝔹) × Val (SelState 𝔹) × Set DVertex
+      linkIntermediates ι =
+         let
+            ια = Env $ ιfromαs g0 (keys ι) :: Env Vertex
+            ι' = ι <#> getSel Transient >>> to𝔹
+            αs = selectαs ι' ια
+            v = ι_fwd αs
+            γ = ι_bwd αs
+         in
+            γ × v × (dvertices g0 αs)
 
    pure
       { spec
       , s
-      , γ: γInert <*> γ0 <*> γ0
-      , v: vInert <*> v0 <*> v0
+      , γ: selStates <$> inert.γ <*> unselected.γ <*> unselected.γ
+      , v: selStates <$> inert.v <*> unselected.v <*> unselected.v
+      , ι: empty
       , linkedOutputs
       , linkedInputs
-      , dir: LinkedOutputs
+      , linkIntermediates
+      , dir: { persistent: LinkedOutputs, transient: LinkedOutputs }
       , in_views
       , out_view: Nothing
       , intermediate_views: empty
-      , intermediate_values: empty
       , in_roots
+      , inerts: inertFwd ∩ inertBwd
       }
 
+dvertices :: forall g. Graph g => g -> Set Vertex -> Set DVertex
+dvertices g αs =
+   (\α -> DVertex (α × (vertexData g α))) `Set.map` αs
+
+ιfromαs :: forall g. Graph g => g -> Set String -> Dict (Val Vertex)
+ιfromαs g αs = D.fromFoldable $ Set.mapMaybe
+   ( \α -> do
+        v <- asVal $ vertexData g (Vertex α)
```

Comment:
Idiomatically I'd probably write this with as map, as in `(α × _) <$>`, dropping to the `do` so you can write the lambda as a one-liner and the omitting the `αs` argument too.